python/wtte/wtte.py

```diff
@@ -166,13 +166,13 @@ def loglik_discrete(y, u, a, b, epsilon=K.epsilon()):
     hazard1 = K.pow((y + 1.0) / a, b)
 
     loglikelihoods = u * \
-        K.log(K.exp(hazard1 - hazard0) - (1.0 - epsilon)) - hazard1
+        K.log((1.0 + epsilon) - K.exp(hazard0 - hazard1)) - hazard0
     return loglikelihoods
 
 
 def loglik_continuous(y, u, a, b, epsilon=K.epsilon()):
     ya = (y + epsilon) / a
-    loglikelihoods = u * (K.log(b) + b * K.log(ya)) - K.pow(ya, b)
+    loglikelihoods = u * (K.log(b) - K.log(a) + (b-1) * K.log(ya)) - K.pow(ya, b)
```

Comment:
Considering the only terms that differ:

```
b * K.log(ya)
vs
- K.log(a) + (b-1) * K.log(ya)
```

Where the latter can be written
```
- K.log(a)+b* K.log(ya)-1* K.log(ya)
=[b* K.log(ya)]- K.log(a)-1* K.log(ya)
=[b* K.log(ya)]- K.log(a)-K.log(y+eps)+K.log(a)
=[b* K.log(ya)]- K.log(y+eps)
```
With regards to the parameters, this is proportional to 
```
∝b * K.log(ya)
```

Since `K.log(y+eps)` has zero gradient, so it's unnecessary to compute it. The standard in almost every statistical/ml package I've seen is - for computational reasons- to implement loss functions using the only the terms that are proportional- rather than equal to the log-likelihood.

The upsides are very marginal computational benefit. The downsides are that it can be confusing if one expects that `exp(-loss)="probability of seeing this data with these parameters"`. Log-likelihood are hence rarely interpretable or directly comparable across distributions and implementations except in proportionality. 

I'm leaning towards that the upsides is not worth the downsides and would be open for a loss function that is equal, rather than proportional. I'm a bit worried about touching these equations too much, they are battle tested in numerous fights against NaN and have proven themselves very stable and performant. A form like
```
     loglikelihoods = u * (K.log(b) + b* K.log(ya)- K.log(y-eps)) - K.pow(ya, b)
```
Seems like a pretty safe and cheap alternative however.

What do you think?

Reply:
Now when I come think of it once again, you're right! There is no need to add redundant constant term to the log-likelihood in continuous case.